### PR TITLE
Doc: Replace matplotlibrc.template

### LINKED ISSUE #25678: [Doc]: matplotlibrc.template does not exist anymore
### Documentation Link

https://matplotlib.org/stable/api/rcsetup_api.html

### Problem

"Any additions or deletions to the parameter set listed here should also be propagated to the matplotlibrc.template in Matplotlib's root source directory."

matplotlibrc.template is now called matplotlibrc and located in lib/matplotlib/mpl-data/

Grepping:
```
lib/matplotlib/__init__.py:https://github.com/matplotlib/matplotlib/blob/%(version)s/matplotlibrc.template
lib/matplotlib/font_manager.py:            matplotlibrc.template file
lib/matplotlib/rcsetup.py:propagated to the :file:`matplotlibrc.template` in Matplotlib's root source
lib/matplotlib/rcsetup.py:# The rcParams defaults are defined in matplotlibrc.template, which gets copied
lib/matplotlib/rcsetup.py:_hardcoded_defaults = {  # Defaults not inferred from matplotlibrc.template...
lib/matplotlib/text.py:            :ref:`matplotlibrc.template file
```

### Suggested improvement

Update all references to matplotlibrc.template, except for those in old change notes:
```
doc/users/prev_whats_new/changelog.rst:    matplotlibrc.template so that text will be scaled by default when changing
doc/users/prev_whats_new/changelog.rst:    commented out all default values in matplotlibrc.template Default values
doc/users/prev_whats_new/github_stats_3.1.1.rst:* :ghpull:`14409`: Backport PR #14335 on branch v3.1.x (Add explanation of animation.embed_limit to matplotlibrc.template)
doc/users/prev_whats_new/github_stats_3.1.1.rst:* :ghpull:`14335`: Add explanation of animation.embed_limit to matplotlibrc.template
doc/users/prev_whats_new/github_stats_3.2.0.rst:* :ghpull:`15108`: Fix doc build and resync matplotlibrc.template with actual defaults.
doc/users/prev_whats_new/github_stats_3.2.0.rst:* :ghpull:`14723`: Cleanup some parameter descriptions in matplotlibrc.template
doc/users/prev_whats_new/github_stats_3.2.0.rst:* :ghpull:`14709`: Try to improve the readability and styling of matplotlibrc.template file
doc/users/prev_whats_new/github_stats_3.2.0.rst:* :ghpull:`14671`: Improve readability of matplotlibrc.template
doc/users/prev_whats_new/github_stats_3.3.0.rst:* :ghpull:`15029`: Get default params from matplotlibrc.template.
doc/users/prev_whats_new/github_stats_3.3.0.rst:* :ghpull:`16523`: Reorder mathtext rcparams in matplotlibrc template.
doc/users/prev_whats_new/github_stats_3.3.0.rst:* :ghpull:`15023`: Simplify formatting of matplotlibrc.template.
doc/users/prev_whats_new/github_stats_3.5.0.rst:* :ghpull:`19603`: Remove matplotlibrc.template.
doc/users/prev_whats_new/github_stats_3.5.0.rst:* :ghpull:`19718`: Small style fixes to matplotlibrc.template.
```
(The above should **BE KEPT**.)

### PR DESCRIPTION
**Pull request by SnorfYang**
## PR Summary
Close #25678

### CODE REVIEW

**Comment by SnorfYang** [comment]
cc @rcomer @ksunden @tacaswell

**Comment by rcomer** [comment]
Thanks @SnorfYang, it looks like the documentation build failed due to #25731.  So I’ve just re-spun that.

**Comment by SnorfYang** [comment]
> Thanks @SnorfYang, it looks like the documentation build failed due to #25731. So I’ve just re-spun that.

Now it passes.

File: lib/matplotlib/text.py [hunk 1]
```diff
@@ -1103,7 +1103,7 @@ def set_math_fontfamily(self, fontfamily):
             The name of the font family.
 
             Available font families are defined in the
-            :ref:`matplotlibrc.template file
+            :ref:`lib/matplotlib/mpl-data/matplotlibrc file
```
Comment (rcomer) [inline lib/matplotlib/text.py:1106]: Similar to the font_manager.py change, I think including the Github path here would just confuse the user.
https://output.circle-artifacts.com/output/job/be496fb6-3b21-4e89-83df-1e83d5755dcd/artifacts/0/doc/build/html/api/text_api.html#matplotlib.text.Text.set_math_fontfamily

File: lib/matplotlib/font_manager.py [hunk 2]
```diff
@@ -878,7 +878,7 @@ def set_math_fontfamily(self, fontfamily):
             The name of the font family.
 
             Available font families are defined in the
-            matplotlibrc.template file
+            lib/matplotlib/mpl-data/matplotlibrc file
```
[the remark this answers is not in the document]
Comment (SnorfYang) [reply lib/matplotlib/font_manager.py]: > This docstring now renders like this: https://output.circle-artifacts.com/output/job/be496fb6-3b21-4e89-83df-1e83d5755dcd/artifacts/0/doc/build/html/api/font_manager_api.html#matplotlib.font_manager.FontProperties.set_math_fontfamily
> 
> I do not think including the GitHub path helps the user here, so I would remove that entirely. Also the use of “here” for the link text is quite outdated, so you could update that with a description of what is being linked.

Done.

File: lib/matplotlib/font_manager.py [hunk 3]
```diff
@@ -877,9 +877,8 @@ def set_math_fontfamily(self, fontfamily):
         fontfamily : str
             The name of the font family.
 
-            Available font families are defined in the
-            lib/matplotlib/mpl-data/matplotlibrc file
-            :ref:`here <customizing-with-matplotlibrc-files>`
+            See
+            :ref:`available font families <customizing-with-matplotlibrc-files>`.
```
Comment (rcomer) [inline lib/matplotlib/font_manager.py:881]: I think this description would now make the user expect the link to take them straight to a list of font families.  In fact, they have to scroll down and search for it in the matplotlibrc comments.  So I think the original was mostly right.  Maybe something like “Available font families are defined in the default matplotlibrc file”?

**Review by rcomer — APPROVED** [review]
Looks good to me :+1:

**Comment by rcomer** [comment]
Thank you for your contribution @SnorfYang and congratulations on your first PR in Matplotlib!  We hope to hear from you again.

**Comment by rcomer** [comment]
@meeseeksdev backport to v3.7.x

**Pull request merged**